packages/visx-shape/src/util/getSplitLineSegments.ts

```diff
@@ -1,80 +1,110 @@
-
-    // create a single path element if not done already
-    if (!pathElement) {
-      const svg = document.createElementNS(SVG_NAMESPACE_URL, 'svg');
-      // not visible
-      svg.style.opacity = '0';
-      svg.style.width = '0';
-      svg.style.height = '0';
-      // off screen
-      svg.style.position = 'absolute';
-      svg.style.top = '-100%';
-      svg.style.left = '-100%';
-      // no mouse events
-      svg.style.pointerEvents = 'none';
-      pathElement = document.createElementNS(SVG_NAMESPACE_URL, 'path');
-      pathElement.setAttribute('id', MEASUREMENT_ELEMENT_ID);
-      svg.appendChild(pathElement);
-      document.body.appendChild(svg);
-    }
-
+    const pathElement = getOrCreateMeasurementElement(MEASUREMENT_ELEMENT_ID);
     pathElement.setAttribute('d', path);
+    const totalLength = pathElement.getTotalLength();
 
-    const totalPathLength = pathElement.getTotalLength();
-    const totalPieces = segments.reduce((sum, curr) => sum + curr.length, 0);
-    const pieceSize = totalPathLength / totalPieces;
+    const numSegments = pointsInSegments.length;
+    const lineSegments: LineSegments = pointsInSegments.map(() => []);
 
-    let cumulativeSize = 0;
+    if (segmentation === 'x' || segmentation === 'y') {
+      const segmentBegins = pointsInSegments.map(
+        points => points.find(p => typeof p[segmentation] === 'number')?.[segmentation],
+      );
 
-    const lineSegments = segments.map(segment => {
-      const segmentPointCount = segment.length;
-      const coords: { x: number; y: number }[] = [];
+      const first = pathElement.getPointAtLength(0);
+      const last = pathElement.getPointAtLength(totalLength);
+      const isIncreasing = last[segmentation] > first[segmentation];
+      const isBeyondSegments = isIncreasing
+        ? segmentBegins.map(begin =>
+            typeof begin === 'undefined' ? TRUE : (xOrY: number) => xOrY >= begin,
+          )
+        : segmentBegins.map(begin =>
+            typeof begin === 'undefined' ? TRUE : (xOrY: number) => xOrY <= begin,
+          );
 
-      for (let i = 0; i < segmentPointCount + sampleRate; i += sampleRate) {
-        const distance = (cumulativeSize + i) * pieceSize;
-        const point = pathElement!.getPointAtLength(distance);
-        coords.push(point);
+      let current = 0;
+      for (let distance = 0; distance <= totalLength; distance += sampleRate) {
+        const sample = pathElement.getPointAtLength(distance);
+        const position = sample[segmentation];
+        while (current < numSegments - 1 && isBeyondSegments[current + 1](position)) {
+          current += 1;
+        }
+        lineSegments[current].push(sample);
       }
+    } else {
+      // segmentation === "length"
+      const numPointsInSegment = pointsInSegments.map(points => points.length);
+      const numPoints = numPointsInSegment.reduce((sum, curr) => sum + curr, 0);
+      const lengthBetweenPoints = totalLength / (numPoints - 1);
```

Comment:
Not sure if this matters/might be an unlikely edge case

```suggestion
      const lengthBetweenPoints = totalLength / Math.max(1, (numPoints - 1));
```